packages/discovery-provider/plugins/pedalboard/apps/trending-challenge-rewards/src/rewards.ts

```diff
@@ -0,0 +1,139 @@
+import { WebClient } from '@slack/web-api'
+import { formatDisbursementTable } from './slack'
+import { discoveryDb } from './utils'
+import { ChallengeId } from '@audius/sdk'
+import axios from 'axios'
+
+type Challenge = {
+  challenge_id: string
+  user_id: string
+  specifier: string
+  amount: string
+  completed_blocknumber: number
+  handle: string
+  wallet: string
+}
+
+
+// TODO: move something like this into App so results are commonplace for handlers
+export const disburseTrendingRewards = async (
+  app: App<SharedData>
+): Promise<void> => {
+  const { dryRun } = app.viewAppData()
+  const disburse = await onDisburse(app, dryRun)
+  disburse.mapErr(console.error)
+}
+
+// dryRun separated out so it can be run manually via another trigger
+export const onDisburse = async (
+  app: App<SharedData>,
+  dryRun: boolean,
+  targetSpecifier?: string
+): Promise<Result<undefined, string>> => {
+  const db = discoveryDb
+  const { slackBotToken, slackChannel } = app.viewAppData()
+  if (slackBotToken === undefined) return new Err('SLACK_BOT_TOKEN undefined')
+  const client = new WebClient(slackBotToken)
+
+  console.log(`doing ${dryRun ? 'a dry run' : 'the real deal'}`)
+
+  const sdk = app.viewAppData().sdk
+
+  let completedBlock, specifier
+  if (!targetSpecifier) {
+    const completedBlockRes = await findStartingBlock(db)
+    if (completedBlockRes.err) return completedBlockRes
+    const [startingBlock, startBlockSpecifier] = completedBlockRes.unwrap()
+    completedBlock = startingBlock
+    specifier = startBlockSpecifier
+  } else {
+    const response = await getTrendingChallengesByDate(db, targetSpecifier)
+    const challenge = response[0]
+    completedBlock = challenge.completed_blocknumber! - 1
+    specifier = challenge.specifier
+  }
+  
+  // Fetch all undisbursed challenges
+  const endpoint = await sdk.services.discoveryNodeSelector.getSelectedEndpoint()
+  console.log('endpoint = ', endpoint)
+  const res = await axios.get(
+    `${endpoint}/v1/challenges/undisbursed?completed_blocknumber=${completedBlock}`
+  )
+  const data: Challenge[] = res.data.data
+  const toDisburse = data.filter((c) =>
+    ['tt', 'tp', 'tut'].includes(c.challenge_id)
+)
+
+// Claim all undisbursed challenges
+for (const challenge of toDisburse) {
+  const challengeId = Object.values(ChallengeId).find(id => id === challenge.challenge_id)!
+  console.log('Claimable challengeId = ', challenge)
+  let res
+  let retries = 10
+  while (retries > 0) {
+    try {
+      if (!dryRun) {
+        res = await sdk.challenges.claimReward({
+          challengeId,
+          userId: challenge.user_id,
+          specifier: challenge.specifier,
+          amount: parseFloat(challenge.amount),
+        })
+        console.log('res = ', res)
+        break // Success - exit retry loop
+      }
+    } catch (e) {
+      console.error(`Error claiming reward, challengeId = ${challengeId}, attempt ${11 - retries} of 10, error = `, e)
+      retries--
```

Comment:
I'm surprised this doesn't cause a lint error. Usually ESLint wants you to use the assigning version (`-=`)